src/utils/QCCalculator.cpp

```diff
@@ -144,11 +147,21 @@ class TOPPQCCalculator :
     String inputfile_consensus = getStringOption_("consensus");
     String inputfile_raw = getStringOption_("in");
     String outputfile_name = getStringOption_("out");
-    String contactName = getStringOption_("name");
-    String contactAddress = getStringOption_("address");
+    String contact_name = getStringOption_("name");
+    String contact_address = getStringOption_("address");
     String description = getStringOption_("description");
     String label = getStringOption_("label");
     bool remove_duplicate_features(getFlag_("remove_duplicate_features"));
+    
+    // ensure output file hase valid extension
+    FileTypes::Type out_type = FileHandler::getConsistentOutputfileType(outputfile_name, getStringOption_("out_type"));
+
+    // exit early if invalid file type
+    if (out_type != FileTypes::QCML or out_type != FileTypes::MZQC)
```

Comment:
not needed. checked by TOPPBASE

Reply:
this covers the case when the user doesn't specify output_type and the output file name has a non-valid extension. In that case I don't want to read the mzML file and exit early. but this should not happen often, so it can be removed.

Reply:
in that case, `getConsistentOutputfileType` will throw an exception for you